specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/vulnerabilityAssessments.json

```diff
@@ -0,0 +1,199 @@
+            }
+        }
+    },
+    "paths": {
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Security/vulnerabilityAssessments": {
+            "get": {
+                "x-ms-examples": {
+                    "Get a list of vulnerability assessments on a resource. Though this API returns a list, Currently Microsoft.Security only supports a single default type of vulnerability assessment which is vulnerability assessment on IaaS VMs.": {
+                        "$ref": "./examples/VulnerabilityAssessments/ListByExtendedResourceVulnerabilityAssessments_example.json"
+                    }
+                },
+                "description": "Gets a list of vulnerability assessment onboarding statuses on a given resource.",
+                "tags": [
+                    "VulnerabilityAssessments"
+                ],
+                "operationId": "VulnerabilityAssessment_ListByExtendedResource",
+                "parameters": [
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+                    },
+                    {
+                        "name": "resourceNamespace",
+                        "in": "path",
+                        "description": "The Namespace of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceType",
+                        "in": "path",
+                        "description": "The type of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceName",
+                        "in": "path",
+                        "description": "Name of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+                    }
+                ],
+                "responses": {
+					"200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/VulnerabilityAssessmentsList"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+                        }
+                    }
+                }
+            },
+			"put": {
```

Comment:
A PUT cannot be performed against a collection Uri. In this URI, `vulnerabilityAssessments` is a type name, not an ID, so this is a collection URI. The way this should be modeled is as a singleton:
`/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Security/vulnerabilityAssessments/default` - where "default" is the standard resource name for singletons

You would add both PUT, DELETE, and GET methods for this singleton URI, but please leave the collection GET in place as well (.../vulnerabilityAssessments).  If you plan to have different types of vulnerabilityAssesments in the future, and do not want to use `default` as the resource identifier, you can use an enum of names (assessmentA, assessmentB).

Reply:
Thanks for the review!

Though it is unlikely that there will be more than a 'default' vulnerability assessment resource, I would not want to block it- so I'll set it as a parameter.